File: backend/test/com/skewwhiffy/auraltester/clefs/ClefUpperLedgerNotesTest.scala
package com.skewwhiffy.auraltester.clefs

import com.skewwhiffy.auraltester.scales.Key
import com.skewwhiffy.auraltester.testutils.TestData
import org.scalatest.flatspec.AnyFlatSpec

class ClefUpperLedgerNotesTest extends AnyFlatSpec {
  it should "correctly define lower ledger notes from clef" in {
    val clef = TestData.noteFactories.clef.treble
    val expectedAbc = List("g", "a", "b", "c'", "d'", "e'")
    val expectedNames = expectedAbc.map(it => it.substring(0, 1).toUpperCase)

    val actual = ClefUpperLedgerNotes(clef)

    assert(actual.notes.map(it => it.abc(Key.cMajor)) == expectedAbc)
    assert(actual.notes.map(it => it.lyric) == expectedNames.map(it => Some(it)))
  }
}

File: backend/src/test/kotlin/com/skewwhiffy/notation/model/abc/ClefUpperLedgerNotesTest.kt
package com.skewwhiffy.notation.model.abc

import com.skewwhiffy.notation.model.Key
import com.skewwhiffy.test.util.TestData
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class ClefUpperLedgerNotesTest {

  @Test
  fun `correctly define lower ledger notes from clef`() {
    val clef = TestData.noteFactories.clef.treble
    val expectedAbc = listOf("g", "a", "b", "c'", "d'", "e'")
    val expectedNames = expectedAbc.map { it.substring(0, 1).uppercase() }

    val actual = ClefUpperLedgerNotes(clef)

    assertThat(actual.notes.map { it.abc(Key.cMajor) }).isEqualTo(expectedAbc)
    assertThat(actual.notes.map { it.lyric }).isEqualTo(expectedNames)
  }
}